src/components/PopupWithForm.js

```diff
@@ -41,4 +46,14 @@ export default class PopupWithForm extends Popup {
       input.value = data[input.name];
     });
   }
+
+  // render loading message
+  renderLoading(isLoading, defaultText) {
```

Comment:
The logic for rendering loading looks great!

Also, you can save the initial text in the constructor instead of passing it via arguments

```js
constructor() {
  ..
  this._buttonText = this._button.textContent;
```

Reply:
Got it.